items/resources.py

```diff
@@ -145,28 +145,39 @@ def prepend_urls(self):
     ]
 
   def login(self, request, **kwargs):
-    if request.method == 'POST':
-      data = json.loads(request.body)
-      username = data['username']
-      password = data['password']
+    self.method_check(request, allowed=['post'])
+    current_user = request.user
+    if current_user.is_anonymous():
+      data = self.deserialize(request, request.body,
+      format=request.META.get('CONTENT_TYPE', 'application/json'))
+      username = data.get('username')
+      password = data.get('password')
       user = auth.authenticate(username=username, password=password)
       if user is not None:
-        auth.login(request, user)
-        response = self.create_response(request, Unpacking.unpack_user(user))
-        return HttpResponse(response)
+        if user.is_active:
+          auth.login(request,user)
+          return self.create_response(request, {'user' : Unpacking.unpack_user(user)})
+        else:
+          return self.create_response(request, {}, HttpForbidden )
       else:
-        return HttpResponse(status=500)
+        return self.create_response(request, {}, HttpUnauthorized)
     else:
-      return HttpResponse(status=500)
+      return self.create_response(request, {'user' : Unpacking.unpack_user(current_user)})
 
   def logout(self, request, **kwargs):
-    auth.logout(request)
-    return HttpResponse(status=200)
+    self.method_check(request, allowed=['get'])
+    if request.user and request.user.is_authenticated():
+      auth.logout(request)
+      return self.create_response(request,{})
+    else:
+      return self.create_response(request,{}, HttpUnauthorized)
 
   def get_user(self, request, **kwargs):
     user = request.user
-    response = self.create_response(request, Unpacking.unpack_user(user))
-    return HttpResponse(response)
+    if user.is_anonymous():
+      return self.create_response(request, {}, HttpUnauthorized)
+    else:
+      return self.create_response(request,{'user' : Unpacking.unpack_user(user)})
```

Comment:
if you look into pep8 or pyflakes you'll see this code is not passed. It's kind of lintering for python